lib/llm-events/langchain/event.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { makeId } = require('../../util/hashes')
+const { DESTINATIONS } = require('../../../lib/config/attribute-filter')
+const { isSimpleObject } = require('../../util/objects')
+
+/**
+ * @typedef {object} LangChainEventParams
+ * @property {object} agent A New Relic agent instance.
+ * @property {object} segment A New Relic segment instance.
+ * @property {string} runId The identifier LangChain has assigned to the "run."
+ * @property {Object<string, string>} metadata The metadata, if any, associated with the
+ * LangChain run.
+ * @property {boolean|undefined} [virtual=true] Indicates that this event is a
+ * LangChain specific event (`true`). LangChain is not itself an LLM, but an
+ * interface to many LLMs. Any LLMs LangChain interacts with that we have
+ * instrumented will have their own traces that are not "virtual."
+ */
+/**
+ * @type {LangChainEventParams}
+ */
+const defaultParams = {
+  agent: {},
+  segment: {
+    transaction: {}
+  },
+  runId: '',
+  metadata: {},
+  virtual: undefined
+}
+
+/**
+ * Baseline object representing an event in a LangChain conversation.
+ */
+class LangChainEvent {
+  id = makeId(36)
```

Comment:
is this always the case? i thought in the case of LlmChatCompletionMessages the id is the id of the `summary - sequence` so it'll be passed in

Reply:
I don't know? https://newrelic.atlassian.net/wiki/spaces/INST/pages/3037003838/Python+Agent+LangChain+Instrumentation has:

| Attr | Summary | Message |
|---|---|---|
| id | yes | yes|

No further explanation is given as to what `id` is or should be. It would be easier to understand if we had a singular doc that outlines what `Event`, `Summary`, and `Message` fields are and how they should be defined.

Reply:
`params.runId` is common to all entities and is mapped to `request_id`. My guess is that the `${response_id}-${sequence}` pattern is to facilitate matching messages to summaries? Or something along those lines? If so, whey aren't we doing something like:

```sql
select *
from messages a
join summaries b
on a.response_id = b.response_id
order by a.sequence asc
```

Reply:
so i misspoke but in all of our other instrumentations the LlmChatCompletionMessage id is `<response id>-<sequence>` (if a unique response id exists) or `<unique-id>`-`<sequence>`.